src/Playwright/Core/AssertionsBase.cs

```diff
@@ -0,0 +1,121 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Playwright.Helpers;
+using Microsoft.Playwright.Transport.Protocol;
+
+namespace Microsoft.Playwright.Core
+{
+    public class AssertionsBase
+    {
+        public AssertionsBase(ILocator actual, bool isNot)
+        {
+            ActualLocator = (Locator)actual;
+            IsNot = isNot;
+        }
+
+        protected bool IsNot { get; private set; }
+
+        internal Locator ActualLocator { get; private set; }
+
+        internal async Task ExpectImplAsync(string expression, ExpectedTextValue textValue, object expected, string message, FrameExpectOptions options)
+        {
+            await ExpectImplAsync(expression, new ExpectedTextValue[] { textValue }, expected, message, options).ConfigureAwait(false);
+        }
+
+        internal async Task ExpectImplAsync(string expression, ExpectedTextValue[] expectedText, object expected, string message, FrameExpectOptions options)
+        {
+            options = options ?? new();
+            options.ExpectedText = expectedText;
+            options.IsNot = IsNot;
+            await ExpectImplAsync(expression, options, expected, message).ConfigureAwait(false);
+        }
+
+        internal async Task ExpectImplAsync(string expression, FrameExpectOptions expectOptions, object expected, string message)
+        {
+            if (expectOptions.Timeout == null)
+                expectOptions.Timeout = 5_000;
+            if (expectOptions.IsNot)
+                message = message.Replace("expected to", "expected not to");
+            var result = await ActualLocator.ExpectAsync(expression, expectOptions).ConfigureAwait(false);
+            if (result.Matches == IsNot)
+            {
+                var actual = result.Received;
+                var log = string.Join("\n", result.Log);
+                if (!string.IsNullOrEmpty(log))
+                    log = "\nCall log:\n" + log;
+                if (expected == null)
+                    throw new PlaywrightException($"{message} {log}");
+                throw new PlaywrightException($"{message} '{FormatValue(expected)}'\nBut was: '{FormatValue(actual)}' {log}");
+            }
+        }
+
+        internal ExpectedTextValue ExpectedRegex(Regex pattern, ExpectedTextValue options = null)
+        {
+            if (pattern == null)
+                throw new ArgumentNullException(nameof(pattern));
```

Comment:
```suggestion
            if (pattern == null)
            {
                throw new ArgumentNullException(nameof(pattern));
            }
```

Reply:
Don't we have a rule for this? 🤓

Reply:
https://github.com/microsoft/playwright-dotnet/issues/2024